### PR TITLE
Basic Compact Interactive List documentation

### PR DESCRIPTION
**Pull request by adoroshk**
### Summary
This PR:

1. Adds a Basic Compact Interactive List example and a new page to the compact interactive list documentation that explains the usage of base props and approach to the basic compact interactive list component creation.
2. Fixes id's in multiple examples.
3. Renamed doc mdx files (without changes) for proper docs order. 

### Testing
This change was tested using:

- [ ] WDIO
- [ ] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

**This PR is part of:**

UXPLATFORM-10114

### CODE REVIEW

File: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+# Basic Compact Interactive List
+
+### Description
+
+* The **[Compact Interactive List](/components/cerner-terra-framework-docs/compact-interactive-list/about)** component consists of unique rows. 
+  Each row has a border, which visually separates it from the other rows in the list.
+* The rows are visually stractured in semantic columns, which do not have visual separators, but provide data alighnment and allow [**keyboard navigation**](/components/cerner-terra-framework-docs/compact-interactive-list/about#keyboard-interactions). 
+  The **[columns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) property** provides the rules for semantic columns structure. 
+* The **[cells](/components/cerner-terra-framework-docs/compact-interactive-list/about#row) property** provides the unique row content.
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx:13]: Does it necessarily have to be unique? By design, most columns will contain similar content in each row.
```suggestion
* The **[cells](/components/cerner-terra-framework-docs/compact-interactive-list/about#row) property** provides the row content. 
```

File: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+# Basic Compact Interactive List
+
+### Description
+
+* The **[Compact Interactive List](/components/cerner-terra-framework-docs/compact-interactive-list/about)** component consists of unique rows. 
+  Each row has a border, which visually separates it from the other rows in the list.
+* The rows are visually stractured in semantic columns, which do not have visual separators, but provide data alighnment and allow [**keyboard navigation**](/components/cerner-terra-framework-docs/compact-interactive-list/about#keyboard-interactions). 
+  The **[columns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) property** provides the rules for semantic columns structure. 
+* The **[cells](/components/cerner-terra-framework-docs/compact-interactive-list/about#row) property** provides the unique row content. 
+  The number of cells must be exactly the same, as the number of elements in **[columns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) array**.
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx:14]:
```suggestion
  The number of cells in a row must be exactly the same, as the number of elements in **[columns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) array**. 
```

File: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  The **[columns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) property** provides the rules for semantic columns structure. 
+* The **[cells](/components/cerner-terra-framework-docs/compact-interactive-list/about#row) property** provides the unique row content. 
+  The number of cells must be exactly the same, as the number of elements in **[columns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) array**. 
+  If a cell is supposed to be empty, the **[content](/components/cerner-terra-framework-docs/compact-interactive-list/about#row) property** must be an empty string.
+
+### Creating Basic Compact Interactive List
+
+#### Basic Columns Property Example
+
+* Break row data into semantic columns in a meaningful way. Each semantic column should have as little elements as possible.
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx:21]:
```suggestion
* Break row data into semantic columns in a meaningful way. Each semantic column should have as few elements as possible. 
```

File: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+### Creating Basic Compact Interactive List
+
+#### Basic Columns Property Example
+
+* Break row data into semantic columns in a meaningful way. Each semantic column should have as little elements as possible. 
+  In the example below, the data is split in following columns: `Icon`, `Service Name`, and `Featured`.
+* Provide unique **[id](/components/cerner-terra-framework-docs/compact-interactive-list/about#column)**
+  and meaningful **[displayName](/components/cerner-terra-framework-docs/compact-interactive-list/about#column)**
+  to all semantic columns. The column headers are hidden, but they will be anounced with the assistive technologies to provide the column context.
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx:25]:
```suggestion
  to all semantic columns. The column headers are hidden, but they will be used by assistive technologies, like screen readers, to provide column context.
```

File: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+* Provide meaningful row headers, as assistive technologies announce the row header cell content per row context change.
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx:26]:
```suggestion
* Provide meaningful, identifying row headers. Assistive technologies use these to provide context as a user navigates through rows.
```

File: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    ],
+  },
+  ... // 4 more row elements
+];
+```
+
+
+#### Basic Compact Interactive List Example
+
+This example demonstrates a basic Compact Interactive List component that consists of 5 semantic rows, organised in 3 semantic columns, where the `Service Name` column is a row header.
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx:80]:
```suggestion
This example demonstrates a basic Compact Interactive List component that consists of 5 semantic rows, organized in 3 semantic columns, where the `Service Name` column is a row header.
```

File: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+``` js
+const DefaultCompactInteractiveList = () => (
+  <CompactInteractiveList
+    id="default-conpact-interactive-list"
+    ariaLabel="Default Compact Interactive List"
+    rows={rows}
+    columns={cols}
+    rowHeaderIndex={1} // points out to Service Name column
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx:89]:
```suggestion
    rowHeaderIndex={1} // Specifies 'Service Name' column as row header for every semantic row.
```

File: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+  />
+);
+```
+
+<BasicCompactInteractiveList />
+
+#### Compact Interactive List with Several Visual Columns
+
+To save space, the rows can be organized in visual columns with **[numberOfColumns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table)
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx:98]:
```suggestion
To save space and facilitate faster keyboard navigation, the rows can be organized in visual columns with **[numberOfColumns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) 
```

**Review by eawww — COMMENTED** [review]
Nice job Alla! Just a few corrections and suggestions.

**Comment by mjpalazzo** [comment]
Great job on the CIL doc @adoroshk!
I noticed one typo in the ARIA section,  rowHeaderIndex, "porperty" should be "property"

**Comment by adoroshk** [comment]
> porperty

Good catch! Fixed in https://github.com/cerner/terra-framework/pull/1984/commits/1ea3548e8a99fde6a4c621ccfa96bca2356a808f

**Review by RachaelMarso — APPROVED** [review]
Edited and approved for OHAIUAD-206303

File: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+
+#### Basic Compact Interactive List Example
+
+This example demonstrates a basic Compact Interactive List component that consists of five semantic rows, organized into three semantic columns, where the `Service Name` column is a row header.
+
+``` js
+const DefaultCompactInteractiveList = () => (
+  <CompactInteractiveList
+    id="default-conpact-interactive-list"
```
Comment (nikhitasharma) [inline packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx:85]:
```suggestion
    id="default-compact-interactive-list"
```

**Review by BenBoersma — APPROVED** [review]

File: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx (hunk 9 above)
Comment (adoroshk) [reply packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx]: Good catch!

**Review by nikhitasharma — APPROVED** [review]

**Pull request merged**